tools/assetpipelineeditor/mainwindow.cpp

```diff
@@ -282,6 +287,69 @@ void MainWindow::viewAll()
     m_camera->viewAll();
 }
 
+void MainWindow::pickEntity(Qt3DRender::QPickEvent *event)
+{
+    using namespace Qt3DCore;
+    using namespace Qt3DRender;
+
+    if (!m_entity || event->button() != QPickEvent::LeftButton)
+        return;
+    CollectionModel *model = qobject_cast<CollectionModel *>(m_ui->collectionBrowser->model());
+    if (!model)
+        return;
+
+#if QT_VERSION >= QT_VERSION_CHECK(5, 13, 0)
+    QEntity *node = event->entity();
+#else
+    auto event_p = QPickEventPrivate::get(event);
+    auto scene_p = QNodePrivate::get(m_entity);
+    QEntity *node = qobject_cast<QEntity *>(scene_p->m_scene->lookupNode(event_p->m_entity));
+#endif
+    if (!node)
+        return;
+
+    if (event->modifiers() == Qt::AltModifier) {
+        while (node) {
+            if (!node->objectName().isEmpty())
+                break;
+            node = node->parentEntity();
+        }
+        if (node) {
+            auto entityIndex = model->index(m_entity->entities(), node->objectName());
+            if (entityIndex.isValid()) {
+                m_ui->collectionBrowser->selectionModel()->select(entityIndex, QItemSelectionModel::ClearAndSelect | QItemSelectionModel::Current);
+                m_ui->collectionBrowser->scrollTo(entityIndex);
+            }
+        }
+    } else {
+#if QT_VERSION >= QT_VERSION_CHECK(5, 13, 0)
```

Comment:
can't componentFromEntity in kuesa_utils_p.h be used here rather than rewriting it?

Reply:
done